addons/sourcemod/scripting/jwp_skin.sp

```diff
@@ -315,28 +316,48 @@ bool IsVipSkinUse(int iClient)
 	return (VIP_IsClientVIP(iClient) && VIP_GetClientFeatureStatus(iClient, g_cVIPFeatureName) == ENABLED);
 }
 
-bool SetModel(int client, bool bSetDefaultModel = false)
+public Action SetModel(Handle timer, int client)
 {
-	if (!g_CvarTRandomSkins.BoolValue && !g_CvarCTRandomSkins.BoolValue) return false; // Disable it , if no random skins
+	bool bSetDefaultModel = false
+
+	if (!g_CvarTRandomSkins.BoolValue && !g_CvarCTRandomSkins.BoolValue)
+	{
+		SkinsCheck = false; // Disable it , if no random skins
+		return Plugin_Handled;
+	}
 	else if (!bSetDefaultModel && g_bVIPExists && IsVipSkinUse(client))
-		return true; // Just exit, so we do not override VIP models
+	{
+		SkinsCheck = true; // Just exit, so we do not override VIP models
+		return Plugin_Continue;
+	}
 	else if (!bSetDefaultModel && IsClientSkinUse(client))
-		return true; // Just exit, so we do not override Shop models
+	{
+		SkinsCheck = true; // Just exit, so we do not override Shop models
+		return Plugin_Continue;
+	}
 	else if (g_cSkin[client][0] != NULL_STRING[0])
 	{
-		if (g_bIsCSGO && !g_bIsSafeToSetModel[0]) return false;
+		if (g_bIsCSGO && !g_bIsSafeToSetModel[0]) 
+		{
+			SkinsCheck = false;
+			return Plugin_Handled;
+		}
 		SetEntityModel(client, g_cSkin[client]);
 		if (g_iSkinId[client] != 0)
 			SetEntProp(client, Prop_Send, "m_nSkin", g_iSkinId[client]);
-		return true;
+		SkinsCheck = true;
+		return Plugin_Continue;
 	}
 	
-	return false;
+	SkinsCheck = false;
+	return Plugin_Handled;
```

Comment:
For what purpose you're returning different values. There are no difference to return, cause your timer is **not repeatable** and doesn't contains any flags.
*Sometimes your code turning me to confuse.*